### DISCUSSION TITLE
Division Area features should have topology rules

### DISCUSSION THREAD

**Discussion by BruceHarold** [OvertureMaps/data#155] [Discussion]
Division Area features should tile the plane.  See my screen capture, the blue dots are places points and I used two division area features to spatially filter the data I extracted.  This is a very common requirement.  However division area feature with id 0850704bbfffffff0156acb0d5ab4971 (London) has an interior void which should be filled by the division area feature with id 085072247fffffff01470c0b730e9c7f (City Of London), but it isn't.  See the resulting gaps around Fleet St, Beech St and Fenchurch St.

![image](https://github.com/OvertureMaps/data/assets/47332097/283976af-28a8-4816-95e8-2f2814fe17cf)

**Comment by stepps00** [comment]
@BruceHarold the London locality (id = 0850704bbfffffff0156acb0d5ab4971) comes from this OSM place, which represents the [ceremonial county of Greater London](https://en.wikipedia.org/wiki/Greater_London): https://www.openstreetmap.org/relation/65606

At the moment, this is expected. Though Overture is actively reviewing divisions area data, specifically at the locality and neighborhood subtypes, so this boundary may change. There are other coverage gaps and overlapping features around London, too, which would ideally be corrected over time.

For now, I would suggest a workaround in your spatial filter. Creating an exterior ring of the London polygon, then converting that ring back to a new polygon could fill that hole you're seeing.

- https://postgis.net/docs/ST_ExteriorRing.html
- https://postgis.net/docs/ST_MakePolygon.html

Something like `ST_MakePolygon(St_ExteriorRing(geometry))`

Running this in DuckDB should result in a GeoJSON file with all of the places data (note: limited attributes) you're interested in.

```
COPY (
  -- Create a temp table with the locality of London
  WITH london AS (
    SELECT 
      id as london_id,
      ST_GeomFromWKB(geometry) as london_geom
    FROM
      read_parquet('s3://overturemaps-us-west-2/release/2024-07-22.0/theme=divisions/type=division_area/*', filename=true, hive_partitioning=1)
    WHERE
      id = '0850704bbfffffff0156acb0d5ab4971'
  ),
  -- Use the geometry of London to filter out places within the locality boundary
  places AS (
    SELECT
      names.primary AS name,
      categories.primary as category,
      ROUND(confidence,2) as confidence,
      ST_GeomFromWKB(geometry) as geometry
    FROM
      read_parquet('s3://overturemaps-us-west-2/release/2024-07-22.0/theme=places/type=*/*', filename=true, hive_partitioning=1)
    INNER JOIN
      london
    ON ST_WITHIN(ST_GeomFromWKB(geometry), ST_MakePolygon(St_ExteriorRing(london.london_geom)))
  )
  -- Export the places selection to a geojson file
  SELECT
    *
  FROM
    places
)
TO
  'london_places.geojson'
WITH (
  FORMAT GDAL,
  DRIVER 'GeoJSON',
  SRS 'EPSG:4326'
);
```

I know this isn't a perfect solution, but should give you a workaround with the results you're looking for.

<img width="1244" alt="Screenshot 2024-08-14 at 10 17 41 AM" src="https://github.com/user-attachments/assets/1afa2e61-0024-46d0-bd5e-6ec1d770b7f7">
